### PR TITLE
[luci/pass] Add unit tests for layer_params_set

### PR DESCRIPTION
**Pull request by seanshpark**
This will add some unit tests for layer_params_set() and quantize.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
for #11280 , from draft  #11513

File: compiler/luci/pass/src/CircleQuantizer.test.cpp [hunk 1]
```diff
@@ -189,3 +193,240 @@ TEST(CircleQuantizerTest, quantize_requant_output_NEG)
...
+
+  luci::CircleInput *input = nullptr;
+  luci::CircleOutput *output = nullptr;
+  luci::CircleConv2D *conv2d1 = nullptr;
+  luci::CircleConv2D *conv2d2 = nullptr;
+  luci::CircleConst *filter = nullptr;
+  luci::CircleConst *bias = nullptr;
+};
+
+void SimpleQuantGraph::init()
```
Comment (jinevening) [inline compiler/luci/pass/src/CircleQuantizer.test.cpp:211]:
```suggestion
// Have two conv layers named "c1" and "c2".
void SimpleQuantGraph::init()
```

File: compiler/luci/pass/src/CircleQuantizer.test.cpp [hunk 2]
```diff
@@ -189,3 +193,240 @@ TEST(CircleQuantizerTest, quantize_requant_output_NEG)
...
+  }
+
+  SimpleCircleQuantizer scq;
+  auto options = scq.init();
+  options->layer_params_set(lpss);
+
+  EXPECT_THROW(scq.quantize(&sqg.g), std::runtime_error);
+}
+
+TEST(CircleQuantizerTest, quantize_layer_param_set_dup2_NEG)
```
Comment (jinevening) [inline compiler/luci/pass/src/CircleQuantizer.test.cpp:398]: IIUC, this tests the case for multiple valid `LayerParams`.

```suggestion
TEST(CircleQuantizerTest, multiple_valid_layer_param_set_NEG)
```

File: compiler/luci/pass/src/CircleQuantizer.test.cpp [hunk 3]
```diff
@@ -189,3 +193,240 @@ TEST(CircleQuantizerTest, quantize_requant_output_NEG)
...
+  }
+
+  SimpleCircleQuantizer scq;
+  auto options = scq.init();
+  options->layer_params_set(lpss);
+
+  EXPECT_THROW(scq.quantize(&sqg.g), std::runtime_error);
+}
+
+TEST(CircleQuantizerTest, quantize_layer_param_set_dup1_NEG)
```
Comment (jinevening) [inline compiler/luci/pass/src/CircleQuantizer.test.cpp:356]:
```suggestion
TEST(CircleQuantizerTest, duplicate_name_in_layerparams_NEG)
```

File: compiler/luci/pass/src/CircleQuantizer.test.cpp [hunk 4]
```diff
@@ -189,3 +193,240 @@ TEST(CircleQuantizerTest, quantize_requant_output_NEG)
...
+  }
+
+  SimpleCircleQuantizer scq;
+  auto options = scq.init();
+  options->layer_params_set(lpss);
+
+  EXPECT_NO_THROW(scq.quantize(&sqg.g));
+}
+
+TEST(CircleQuantizerTest, quantize_layer_param_set_no_NEG)
```
Comment (jinevening) [inline compiler/luci/pass/src/CircleQuantizer.test.cpp:320]:
```suggestion
TEST(CircleQuantizerTest, invalid_layer_params_NEG)
```

**Review by jinevening — DISMISSED** [review]
LGTM with minor comments

File: compiler/luci/pass/src/CircleQuantizer.test.cpp (hunk 2 above)
Comment (seanshpark) [reply compiler/luci/pass/src/CircleQuantizer.test.cpp:399]: ah, `valid` may give confusion. how about just `multiple_layer_param_set_NEG` ?

Comment (jinevening) [reply compiler/luci/pass/src/CircleQuantizer.test.cpp:399]: To describe why this is a negative test, how about `duplicate_layer_params_NEG `?

File: compiler/luci/pass/src/CircleQuantizer.test.cpp [hunk 5]
```diff
@@ -189,3 +193,241 @@ TEST(CircleQuantizerTest, quantize_requant_output_NEG)
...
+  }
+
+  SimpleCircleQuantizer scq;
+  auto options = scq.init();
+  options->layer_params_set(lpss);
+
+  EXPECT_THROW(scq.quantize(&sqg.g), std::runtime_error);
+}
+
+TEST(CircleQuantizerTest, quantize_layer_param_set_dup2_NEG)
```
Comment (seanshpark) [inline compiler/luci/pass/src/CircleQuantizer.test.cpp:399]:
```suggestion
TEST(CircleQuantizerTest, duplicate_layer_params_NEG)
```

**Review by jinevening — APPROVED** [review]
LGTM Thank you

**Review by shs-park — COMMENTED** [review]
LGTM!

**Pull request merged**